### PR TITLE
Adds WebVR + Three.js sample

### PR DESCRIPTION
**Pull request by paullewis**
Adds a Hello World Three.js sample for WebVR. Most of the interesting stuff is in the demo-vr.js file :)

PTAL @toji @wibblymat @surma

### CODE REVIEW

File: web-vr/hello-world/demo/demo-vr.js [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      frameData: new VRFrameData()
+    };
+
+    this._addVREventListeners();
+    this._getDisplays();
+  }
+
+  _addVREventListeners () {
+    window.addEventListener('vrdisplayactivate', evt => {
+      evt.preventDefault();
```
Comment (surma) [inline web-vr/hello-world/demo/demo-vr.js:45]: Is the `preventDefault()` actually still necessary?

**Review by surma — COMMENTED** [review]
One Q, otherwise LGTM.

File: web-vr/hello-world/demo/demo-vr.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        'Disable VR' :
+        'Enable VR';
+  }
+
+  _deactivateVR () {
+    if (!this._vr.display) {
+      return;
+    }
+
+    this._vr.display.exitPresent();
```
Comment (wibblymat) [inline web-vr/hello-world/demo/demo-vr.js:103]: 1. You return a `Promise` from this method, but I don't know why
1. `activateVR` does NOT return a `Promise`, even though it's similar
1. `exitPresent` and `requestPresent` return `Promise`s, if you really want to return one.
1. Caveat - `exitPresent` will reject if you weren't actually presenting

File: web-vr/hello-world/demo/demo-vr.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+      this._scene.matrixAutoUpdate = true;
+
+      return super._render();
+    }
+
+    const EYE_WIDTH = this._width * 0.5;
+    const EYE_HEIGHT = this._height;
+
+    // Get all the latest data from the VR headset and dump it into frameData.
+    this._vr.display.getFrameData(this._vr.frameData);
```
Comment (wibblymat) [inline web-vr/hello-world/demo/demo-vr.js:146]: I don't see a submitFrame call... isn't that needed to make the matrices update? I believe that this is somewhere that the polyfill and extension differ from the spec, so check in a real device.

**Review by wibblymat — COMMENTED** [review]
Only reviewed demo-vr.js

File: web-vr/hello-world/demo/demo-vr.js (hunk 1 above)
Comment (paullewis) [reply web-vr/hello-world/demo/demo-vr.js:45]: Removed. AFAIK it's a noop.

File: web-vr/hello-world/demo/demo-vr.js (hunk 3 above)
Comment (paullewis) [reply web-vr/hello-world/demo/demo-vr.js:146]: Fixed

File: web-vr/hello-world/demo/demo-vr.js (hunk 2 above)
Comment (paullewis) [reply web-vr/hello-world/demo/demo-vr.js:103]: Fixed

**Pull request merged**